### PR TITLE
[CARBONDATA-1033] Fixed complex data type in CarbonSource

### PR DESCRIPTION
**Pull request by kunal642**
Analysis: Earlier the flow to get TableModel was different for CarbonSource which cause the create table to fail in case of complex types.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1970/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1971/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala [hunk 1]
```diff
@@ -20,7 +20,7 @@ package org.apache.carbondata.spark.util
 import java.util
 
 import scala.collection.JavaConverters._
-import scala.collection.mutable.Map
+import scala.collection.mutable
```
Comment (ravipesala) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala]: Please don't change file unnecesarily

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1973/

**Comment by ravipesala** [comment]
LGTM

**Pull request merged**